### ISSUE TITLE
FN — a setter call that is not in statement position is invisible to the writer relation

### ISSUE THREAD

**Issue by rboudrouss** [rboudrouss/reactant-analyzer#130]
Found while building the `calls` relation (#126): its first cut inherited the same blindness, and fixing it there made the asymmetry visible.

## The repro

```tsx
export function RenderNested({ wrap }) {
  const [n, setN] = useState(0);
  wrap(setN(1));          // silent
  return <div>{n}</div>;
}

export function RenderDirect() {
  const [n, setN] = useState(0);
  setN(1);                // error  setter-in-render
  return <div>{n}</div>;
}
```

`RenderDirect` reports `setter-in-render` at **Error**. `RenderNested` reports nothing at all — not a downgrade, no row in the writer relation, no `analysis-limit`. This is the false-negative class the project forbids.

## Why

`SetterWalk::expr` runs its machinery only on a `Call` **in statement position**, and its argument loop descends only `FnLit` and `Var` arguments (to resolve callbacks). A `Call` nested anywhere else — another call's argument, a ternary arm, an object field, a JSX prop, a `BinOp` — is never reached, so the write is not in `slot_writers` and no consumer of the relation can see it.

Every reader of the writer relation inherits it: `setter-in-render`, `infinite-loop`, `derived-state`, `redundant-set-state`, the `writers` edge and `writer_phases` in Tier A.

## What already exists

ADR-036 §4 / ADR-037 §4 built exactly the traversal this needs, for the `calls` and `reads` channels: `expr` visits every node once and runs the call machinery on each `Call` it passes, skipping `FnLit` children because the machinery is the only place that knows what class the function runs in. It is gated behind `SetterWalk::scanning()` **solely** so this change could be verified separately — the gate is the scope decision, not a design one.

## Fix direction

Drop the gate. Then measure:

- the corpus must move **up or not at all** — a lost row would mean the traversal reaches a site the old walk reached differently, which would be a bug in the traversal, not a fix;
- watch the runtime: the walk now visits every expression node, and #86 was a perf hang;
- expect new findings, and triage them: a setter reached through a nested call is genuinely a write, but the rules downstream have never seen these rows and may need the phase they arrive with checked (a nested call in a JSX prop is a handler position, not a render write).

## Acceptance

- [ ] `wrap(setN(1))` in a render body reports `setter-in-render`
- [ ] the gate is gone, so the setter, `calls` and `reads` channels share one traversal
- [ ] corpus diff published, nothing removed
- [ ] a regression test that fails when the traversal is reverted

**Comment by rboudrouss** [comment]
Fixed in 7607ac9 — ADR-038.

The gate is gone: `expr` visits every node exactly once and runs the call machinery on each `Call` it passes, skipping `FnLit` children. All three channels (setters, `calls`, `reads`) share one traversal.

**Acceptance**

- [x] `wrap(setN(1))` in a render body reports `setter-in-render` — `tests/fixtures/setter_phase/App.tsx`, and so does a call in a JSX prop value.
- [x] the gate is gone, so the setter, `calls` and `reads` channels share one traversal.
- [x] corpus diff published, nothing removed.
- [x] a regression test that fails when the traversal is reverted — `tests/setter_phase.rs`, six tests, each gated by removing its own fix.

**Corpus** (34,730 files): 6322 → 6340. Runtime unchanged (twenty: 276.0s baseline vs 274.6s, back to back).

**27 added, every one a Warning, no new Errors.** 25 are the ⊤ class: a setter handed to a callee with no timing summary (`handleSubmit(onSubmit)`, `form.onSubmit(cb)`, `composeEventHandlers(a, cb)`). Sound — ⊤ includes the render pass — and now honestly worded. Narrowing them needs #94, not a change here.

**9 removed, none of them a finding**: 4 were a duplicate component heading, 4 came from the redundant props = ⊤ pass on a duplicated component, and 1 was a false positive.

**Two precision defects the wider reach exposed**, both fixed centrally (the issue predicted the first):

1. `setter-in-render` never read the phase the walk assigns every row. A `Handler` or `Deferred` row is a *proof* the write is not in the render pass and is now silent; `Unknown` is ⊤ and stays. `SetterCallPhase` had collapsed `Deferred` and `Unknown` into `Other` — exactly the distinction needed — so they are separate variants now. Error takes two musts: the phase is `Sync`, and calling the variable provably writes (`SetterProp::must_write`, answered by walking the closure's own body rather than guessing from the binding's shape).

2. `ComponentIR.name` was the bare name while the program result is keyed by `display_name`. The moment two files define `Demo`, a component read as its own parent and `cross-setter-in-render` fired at **Error** on it (mantine's `Demo@…`, dub's `TokensPage@…`). The registry now decides one spelling where it hands the IR to the analysis, and the child path in `eval_comp_app` resolves through the same place.

Left open, as the issue anticipated: a row whose statement carries no span still reports without a line (#131), and `collect_component_setter_vars` still attributes an owner from any block env rather than the call site's (#119) — `must_write` narrows what that costs, it does not answer it.
